### PR TITLE
Add bottom sheet menu with settings for theme and animation speed

### PR DESCRIPTION
**Pull request by Renzo-Olivares**
This change adds a modal bottom sheet menu to Reply. The bottom sheet includes settings for changing the app theme, and changing the animation speed. It can be accessed by clicking on the settings icon, while the bottom drawer is open.

`EmailStore`
* Added `_currentTheme` property with getters/setters to keep track of global theme
* Added `_currentAnimationSpeed` property with getters/setters to keep track of global animation speed

`BottomSheetMenu`
* Implemented modal bottom sheet menu with two sections for settings: theme, animation speed
* Implemented `AnimationSpeedSetting` enum and extension.
* Implemented extension for `ThemeMode` to return a String with the current mode.
* Use `ExpansionTile` and `RadioListTile` to create settings menu

`MobileNav`
* Wrap `HomePage` body in a `WillPopScope` to handle back button

`Root`
* Wrap our root `MaterialApp` with a `Selector` to listen for theme changes

reply-bottom-menu

### CODE REVIEW

File: lib/home.dart [hunk 1]
```diff
@@ -432,7 +473,18 @@ class _BottomAppBarActionItems extends StatelessWidget {
...
                     color: ReplyColors.white50,
-                    onPressed: () {},
+                    onPressed: () async {
+                      drawerController.reverse();
+                      showModalBottomSheet(
+                        context: context,
+                        shape: RoundedRectangleBorder(
+                          borderRadius: modalBorder,
+                        ),
+                        builder: (context) {
```
Comment (guidezpl) [inline lib/home.dart:483]: nit: ⇒

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Comment by guidezpl** [comment]
I think the Settings button may benefit from increased visibility by being on the bottom app bar itself

File: lib/bottom_menu.dart [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+class BottomSheetMenu extends StatefulWidget {
+  const BottomSheetMenu();
+
+  @override
+  _BottomSheetMenuState createState() => _BottomSheetMenuState();
+}
+
+class _BottomSheetMenuState extends State<BottomSheetMenu> {
+  AnimationSpeedSetting _animationSpeedSetting;
+  ThemeMode _theme;
```
Comment (rami-a) [inline lib/bottom_menu.dart:55]: rename to `_themeMode` since theme usually means the actual themedata object

File: lib/bottom_menu.dart [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+          color: Theme.of(context).navigationRailTheme.backgroundColor,
+          borderRadius: modalBorder,
+        ),
+        child: SingleChildScrollView(
+          child: Column(
+            children: [
+              ExpansionTile(
+                title: Text('Theme'),
+                children: [
+                  for (var theme in ThemeMode.values)
```
Comment (rami-a) [inline lib/bottom_menu.dart:101]:
```suggestion
                  for (var themeMode in ThemeMode.values)
```

File: lib/bottom_menu.dart [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        break;
+      case ThemeMode.dark:
+        return 'Dark';
+        break;
+    }
+    return null;
+  }
+}
+
+class BottomSheetMenu extends StatefulWidget {
```
Comment (rami-a) [inline lib/bottom_menu.dart:46]: Rename this to something like `SettingsBottomSheet` and maybe rename the file too.

File: lib/bottom_menu.dart [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        Provider.of<EmailStore>(context, listen: false).themeMode = theme;
+      }
+
+      void setAnimationSpeed(AnimationSpeedSetting animationSpeed) {
+        state(() {
+          _animationSpeedSetting = animationSpeed;
+        });
+        Provider.of<EmailStore>(context, listen: false).animationSpeed =
+            animationSpeed;
+        timeDilation = animationSpeed.value;
```
Comment (rami-a) [inline lib/bottom_menu.dart:87]: I wonder if this should be moved into the `EmailStore` to mimic the `notifyListeners` behavior there

**Comment by Renzo-Olivares** [comment]
> I think the Settings button may benefit from increased visibility by being on the bottom app bar itself

I think we can leave it there. In the codelab there should be a section explaining how to slow down animations and change the theme. This also keeps it consistent with the android version.

File: lib/home.dart [hunk 6]
```diff
@@ -308,104 +309,144 @@ class _HomePageState extends State<HomePage> with TickerProviderStateMixin {
...
+          ),
+          bottomNavigationBar: Selector<EmailStore, bool>(
+            selector: (context, emailStore) => emailStore.onMailView,
+            builder: (context, onMailView, child) {
+              _bottomAppBarController.forward();
+
+              return SizeTransition(
+                sizeFactor: _bottomAppBarCurve,
+                axisAlignment: -1,
+                child: BottomAppBar(
```
[the remark this answers is not in the document]
Comment (Renzo-Olivares) [reply lib/home.dart:328]: I agree, will do.

File: lib/home.dart [hunk 7]
```diff
@@ -308,104 +309,144 @@ class _HomePageState extends State<HomePage> with TickerProviderStateMixin {
...
-            : const Padding(
-                padding: EdgeInsetsDirectional.only(bottom: 8),
-                child: _ReplyFab(),
-              ),
-        floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked,
       ),
     );
   }
+
+  Future<bool> _willPopCallback() async {
```
[the remark this answers is not in the document]
Comment (Renzo-Olivares) [reply lib/home.dart:403]: Will do.

**Pull request merged**